apscheduler/triggers/cron/__init__.py

```diff
@@ -194,6 +194,10 @@ def get_next_fire_time(self, previous_fire_time, now):
         if fieldnum >= 0:
             if self.jitter is not None:
                 next_date = self._apply_jitter(next_date, self.jitter, now)
+
+                if self.end_date and next_date > self.end_date:
+                    return None
```

Comment:
No. Applying jitter should not discard an otherwise fine fire time. If you want to fix this, limit the datetime by end_date after applying jitter but don't discard it.